Taskfile.yml

```diff
@@ -75,51 +68,71 @@ tasks:
+        sh: 'echo {{default "$(gopass show cogito/docker_username)" .DOCKER_USERNAME}}'
+      DOCKER_TOKEN:
+        sh: 'echo {{default "$(gopass show cogito/docker_token)" .DOCKER_TOKEN}}'
 
-  docker-build:
-    deps: [docker-login]
+  docker:build:
     desc: Build the Docker image.
     cmds:
-      - docker build --build-arg BUILD_INFO --tag $DOCKER_IMAGE .
+      - docker build --build-arg BUILD_INFO --tag {{.DOCKER_FULL_NAME}} .
     env:
       BUILD_INFO: "{{.BUILD_INFO}}"
 
-  docker-smoke:
+  docker:smoke:
     desc: Simple smoke test of the Docker image.
     cmds:
-      - echo '{{.INPUT}}' | docker run --rm --interactive cogito /opt/resource/check
+      - echo '{{.INPUT}}' | docker run --rm --interactive {{.DOCKER_FULL_NAME}} /opt/resource/check
       - echo
-      - echo '{{.INPUT}}' | docker run --rm --interactive cogito /opt/resource/in dummy
+      - echo '{{.INPUT}}' | docker run --rm --interactive {{.DOCKER_FULL_NAME}} /opt/resource/in dummy
       - echo
     vars:
       INPUT: '{ "source": { "owner": "foo", "repo": "bar", "access_token": "123"} }'
 
-  docker-push:
-    deps: [docker-login]
+  docker:push:
+    desc: Push the Docker image
     cmds:
-      - docker tag $DOCKER_IMAGE $DOCKER_ORG/$DOCKER_IMAGE:$DOCKER_TAG
-      - echo Pushing $DOCKER_ORG/$DOCKER_IMAGE:$DOCKER_TAG
-      - docker push $DOCKER_ORG/$DOCKER_IMAGE:$DOCKER_TAG
+      - docker push {{.DOCKER_FULL_NAME}}
       - |-
         if [ -n "$DOCKER_LATEST" ]; then
-            docker tag $DOCKER_IMAGE $DOCKER_ORG/$DOCKER_IMAGE:$DOCKER_LATEST
-            echo Pushing $DOCKER_ORG/$DOCKER_IMAGE:$DOCKER_LATEST
-            docker push $DOCKER_ORG/$DOCKER_IMAGE:$DOCKER_LATEST
+            docker tag {{.DOCKER_FULL_NAME}} {{.DOCKER_BASE_NAME}}:latest
+            # FIXME safety while adding GH actions: disable pushing "latest"
+            exit 1
+            docker push {{.DOCKER_BASE_NAME}}:latest
+            echo Pushed a new \"latest\" tag!
+        else
+            echo Did not push a new \"latest\" tag.
         fi
-    env:
-      DOCKER_ORG:
-        sh: echo ${DOCKER_ORG:-$(gopass show cogito/docker_org)}
-      DOCKER_TAG:
-        sh: ci/git-ref-to-docker-tag.sh
+    vars:
       DOCKER_LATEST:
         sh: ci/git-latest.sh
+
+  ci:setup:
+    desc: Useful only when running under CI.
+    cmds:
+      - task: install:deps
+      # Running "go mod download" is optional, since "go build" would do it anyway.
+      # We run it explicitly to make the output of "go build" more focused.
+      - cmd: go mod download -x
+
+  # When using GitHub Actions, add this snippet at the end of the workflow:
+  #    - run: docker logout
+  #    # Always remove credentials, also if any previous step failed.
+  #    if: always()
+  ci:teardown:
+    desc: ALWAYS run this when in CI (reduces security exposures)
+    cmds:
+      # Remove credentials from the file system, added by "docker login" :-(
+      - cmd: docker logout
```

Comment:
A _loooong_ story. This is the best we can do currently on GitHub Actions.
It is not bad as it sounds since the Actions VM is transient (like a Concourse task for Linux).